### DISCUSSION TITLE
References on strategy used in DynamicFewShotGPTClassifier

### DISCUSSION THREAD

**Discussion by gajomi** [iryna-kondr/scikit-llm#88] [Q&A]
Is there any literature describing the strategy used by  DynamicFewShotGPTClassifier? The methodology is laid bare in the code itself, but I am interested in comparing to any related/prior art to understand tradeoffs. My main interest is in optimizing for total inference cost including the index creation (as it might compare to training cost to fine tuning on the same data). Also, I am interested to see any extant descriptions of performance on classes with long tail probability distributions (my though it that with a good kernel this approach might be able to do better than fine tuning). Appreciate any links y'all can drop here.

**Comment by OKUA1** [comment]
Hi @gajomi,

At the time `DynamicFewShotGPTClassifier` was implemented, we were not aware of any papers describing the method in that exact formulation. To be honest I would be interested in such a comparison myself.
